### PR TITLE
Use appropriate action bar titles (Fixed #1567)

### LINKED ISSUE #1567: Appropriate title should be visible on the action bar
**Actual Behaviour**

The lablel on the group wise skills listing fragment is SUSI.AI skills.

**Expected Behaviour**

The group name should be displayed instead of SUSI.AI skills.

**Steps to reproduce it**

Open the skills listing fragment, click on any group at the bottom of the skills listing page. Check the title on the action bar.

**Screenshots of the issue**

categoryskill

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by arundhati24**
Fixes #1567 

Changes: The title bar now shows the correct titles for the Skills Listing Fragment, Group Wise Skills Listing Fragment and Skill Details Fragment.

**Screenshots**

screenshot_2018-07-21-09-45-18-028_ai susi

screenshot_2018-07-21-09-45-29-430_ai susi

screenshot_2018-07-21-09-45-20-115_ai susi

### CODE REVIEW

**Comment by batbrain7** [comment]
@arundhati24 did you check my comment on the issue?

**Comment by arundhati24** [comment]
Already replied you there.

File: app/src/main/java/org/fossasia/susi/ai/skills/groupwiseskills/GroupWiseSkillsFragment.kt [hunk 1]
```diff
@@ -43,6 +44,7 @@ class GroupWiseSkillsFragment : Fragment(), IGroupWiseSkillsView, SwipeRefreshLa
         val argument = arguments
         if (argument != null) {
             this.skills.group = argument.getString("group")
+            (activity as SkillsActivity).setActionBarTitle(this.skills.group)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/skills/groupwiseskills/GroupWiseSkillsFragment.kt]: And if in future, `GroupWiseSkillsFragment.kt` is attached to a different Activity?

This is type unsafe and will lead to problems in future

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt [hunk 2]
```diff
@@ -83,6 +83,11 @@ class SkillDetailsFragment : Fragment(), ISkillDetailsView {
                 SKILL_KEY) as SkillData
         skillGroup = (arguments as Bundle).getString(SKILL_GROUP)
         skillTag = (arguments as Bundle).getString(SKILL_TAG)
+        if (skillData != null) {
+            if (skillData.skillName != null) {
+                (activity as SkillsActivity).setActionBarTitle(skillData.skillName)
+            }
+        }
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt]:
```kotlin
skillData?.skillName?.let { setTitle(it) }
```

File: app/src/main/java/org/fossasia/susi/ai/skills/groupwiseskills/GroupWiseSkillsFragment.kt [hunk 3]
```diff
@@ -49,7 +51,11 @@ class GroupWiseSkillsFragment : Fragment(), IGroupWiseSkillsView, SwipeRefreshLa
 
     @NonNull
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        activity?.title = activity?.getString(R.string.skills_activity)
+        if (activity is SkillsActivity) {
+            this.skills.group?.let { activity?.title = it }
+        } else {
+            Timber.d("activity is not SkillsActivity")
+        }
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/skills/groupwiseskills/GroupWiseSkillsFragment.kt]: Check not needed, every activity has setTitle method

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt [hunk 4]
```diff
@@ -88,8 +89,11 @@ class SkillDetailsFragment : Fragment(), ISkillDetailsView {
 
     @NonNull
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        if (skillData.skillName != null && !skillData.skillName.isEmpty())
-            (activity as SkillsActivity).title = skillData.skillName
+        if (activity is SkillsActivity) {
+            skillData?.skillName?.let { activity?.title = skillData.skillName }
+        } else {
+            Timber.d("activity is not SkillsActivity")
+        }
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt]: Same here

**Pull request merged**